modules/storages/app/common/storages/peripherals/nextcloud_connection_validator.rb

```diff
@@ -93,6 +81,29 @@ def non_oidc_provisioned_user
                                       description: I18n.t("storages.health.connection_validation.oidc_non_oidc_user")))
       end
 
+      def token_usability
+        service = OpenIDConnect::UserTokens::FetchService.new(user: @user)
+
+        result = service.access_token_for(audience: @storage.audience)
+        return None() if result.success?
+
+        error_code = case result.failure
+                     in { code: /token_exchange/ | :unable_to_refresh_token }
```

Comment:
:green_circle: Should we rename this to `unable_to_exchange_token`? I see that it originates from the refresh service, but there it only occurs when we try to exchange instead of a refresh.

And here you decided to cause an error of `oidc_cant_exchange_token` instead of `oidc_cant_refresh_token`, so I guess it could make sense?

Reply:
I was torn at this: this errors comes up if:

- The existing token doesn't have the nextcloud storage audience
- No token exchange is possible.

I'm not sure how to capture that exact case in a name and specific error message. =/